src/foam/u2/view/ReadReferenceView.js

```diff
@@ -79,21 +174,73 @@ foam.CLASS({
 
     function fromProperty(prop) {
       this.SUPER(prop);
+      
       this.prop = prop;
-      var dao = this.ctrl.__subContext__[prop.targetDAOKey];
-      this.permissionEnableLinkCheck().then(() => {
-        if ( dao )
+      
+      // fetch link config properties from where they were provided
+      // (i.e., from reference property or this view)
+      this.enableLink = this.prop.enableLink && this.enableLink;
+      this.controlAccessToDAOSummary = this.prop.controlAccessToDAOSummary || this.controlAccessToDAOSummary;
+      this.menuKeys = this.prop.menuKeys.length > 0 ? this.prop.menuKeys : this.menuKeys;
+
+      this.configLink().then(() => {
+        const dao = this.ctrl.__subContext__[prop.targetDAOKey];
+        if ( dao ) {
           dao.find(this.data).then((o) => this.obj = o);
+        }
       });
     },
 
-    async function permissionEnableLinkCheck() {
-      if ( ! this.auth ) return;
-      let permission = `${this.prop.of.id}.${this.prop.name}.disableRefLink`;
-      permission = permission.toLowerCase();
-      await this.auth.check(this.__subContext__, permission).then( check => {
-        this.enableLink = ! check;
-      })
-    }
+    async function configLink() {
+      /*
+       * Uses the flow chart above to control access to the link
+       * and what is set to this link.
+       */
+
+      // enableLink explicitly set to false?
+      if ( ! this.auth || ! this.enableLink ) {
+        this.enableLink = false;
+        this.linkTo = '';
+        return;
+      }      
+      
+      try {
+        // menus are provided?
+        if ( this.menuKeys.length > 0 ) {
+          // check permissions for menus
+          const permissions = await Promise.all([...this.menuKeys].map(menuId => {
+            return this.auth.check(this.__subContext__, `menu.read.${menuId}`);
+          }));
+
+          const firstAt = permissions.indexOf(true);
+          // can read menu?
+          if ( firstAt > -1 ) {
+            this.enableLink = true;
```

Comment:
is this part necessary? since previous condition `( ! this.auth || ! this.enableLink )`, if this.enableLink == false then it will just return so this.enableLink here has to be true

Reply:
@yij793 Yes you are right, this is not really necessary as 'enableLink would be already true in this case. The reason for explicitly setting it was to clearly show what were set to 'enableLink' and 'linkTo' at each leaf nodes. If you look at the end of each if and else statements, you can see I am doing the same thing.